cabal-install/src/Distribution/Client/CmdInstall.hs

```diff
@@ -540,7 +540,9 @@ installAction flags@NixStyleFlags{extraFlags, configFlags, installFlags, project
         -- BuildOutcomes because we also need the component names
           traverseInstall (installCheckUnitExes InstallCheckInstall) installCfg
   where
-    configFlags' = disableTestsBenchsByDefault configFlags
+    -- Filter `configProgSuffix` out, in order to get the canonical executable
+    -- in the store.
+    configFlags' = (disableTestsBenchsByDefault configFlags){configProgSuffix = NoFlag}
```

Comment:
Wouldn't this likely also apply to the flag `configProgPrefix`?

Reply:
Yes. I updated the code to fix it too.